### PR TITLE
fix: hydration mismatch on Windows

### PR DESCRIPTION
**Pull request by himself65**
Fixes https://github.com/dai-shi/waku/issues/337

### CODE REVIEW

**Comment by himself65** [comment]
CI flaky might related to: https://github.com/actions/download-artifact/issues/249

File: e2e/utils.ts [hunk 1]
```diff
@@ -1,8 +1,14 @@
 import net from 'node:net';
-import { test as basicTest } from '@playwright/test';
+import { expect, test as basicTest } from '@playwright/test';
 import type { ConsoleMessage } from '@playwright/test';
 import type { ChildProcess } from 'node:child_process';
 
+const unexpectedErrors: RegExp[] = [
+  /^You did not run Node.js with the `--conditions react-server` flag/,
+  /^\(node:14372\)/,
+  /^Warning: Expected server HTML to contain a matching/,
```
Comment (himself65) [inline e2e/utils.ts:9]: I noticed that there are some hydration mismatch in current CI

File: packages/waku/src/lib/renderers/html-renderer.ts [hunk 2]
```diff
@@ -68,6 +68,9 @@ const getViteServer = async () => {
 };
 
 const loadServerFileDev = async (fileURL: string) => {
+  if (fileURL.endsWith('waku/dist/client.js')) {
+    return import(WAKU_CLIENT_MODULE_VALUE);
+  }
```
Comment (dai-shi) [inline packages/waku/src/lib/renderers/html-renderer.ts:73]: Is this only for our monorepo setup?

Comment (himself65) [reply packages/waku/src/lib/renderers/html-renderer.ts:73]: I think so, but cannot find why

**Comment by himself65** [comment]
I guess this only happened in Monorepo. So, if it possible we change the way of writing examples?

Or this is still some corner case on Windows

**Comment by himself65** [comment]
Ok, the real bug is `/` on prefix again

**Comment by himself65** [comment]
image
image

File: packages/waku/src/lib/utils/path.ts [hunk 3]
```diff
@@ -36,8 +36,17 @@ export const fileURLToFilePath = (fileURL: string) => {
 };
 
 // for filePath
```
Comment (dai-shi) [inline packages/waku/src/lib/utils/path.ts:38]: joinPath is only for "filePath" style. Please do not modify for multiple path styles. We should adjust it in advance. This makes things more maintainable.

**Review by dai-shi — CHANGES_REQUESTED** [review]

**Review by dai-shi — APPROVED** [review]
LGTM!

**Pull request merged**